client/lib/cart-values/cart-items.js

```diff
@@ -137,9 +139,11 @@ function removeItemAndDependencies( cartItemToRemove, cart ) {
  * @returns {Object[]} the list of dependency items in the shopping cart
  */
 function getDependentProducts( cartItem, cart ) {
-	return getAll( cart ).filter( function( existingCartItem ) {
+	const dependentProducts = getAll( cart ).filter( function( existingCartItem ) {
 		return isDependentProduct( cartItem, existingCartItem );
 	} );
+
+	return uniq( flatten( dependentProducts.concat( dependentProducts.map( dependentProduct => getDependentProducts( dependentProduct, cart ) ) ) ) );
```

Comment:
_Technically_, if we ever get a third level dependency product (Plan -> Domain -> Gapps -> _something_), `flattenDeep` will prevent a regression :wink: Since `flatten` only flattens one level deep.
I know, this is nitpicking, but _just in case_ we do have such a deep dependency ;)